tests/expr_and_series/unique_test.py

```diff
@@ -1,20 +1,49 @@
 from __future__ import annotations
 
-import narwhals.stable.v1 as nw
+from contextlib import nullcontext as does_not_raise
+
+import pytest
+
+import narwhals as nw
+from narwhals.exceptions import LengthChangingExprError
+from tests.utils import Constructor
 from tests.utils import ConstructorEager
 from tests.utils import assert_equal_data
 
 data = {"a": [1, 1, 2]}
 data_str = {"a": ["x", "x", "y"]}
 
 
-def test_unique_expr(constructor_eager: ConstructorEager) -> None:
-    df = nw.from_native(constructor_eager(data))
-    result = df.select(nw.col("a").unique())
-    expected = {"a": [1, 2]}
+def test_unique_expr(constructor: Constructor) -> None:
+    df = nw.from_native(constructor(data))
+    context = (
+        pytest.raises(LengthChangingExprError)
+        if isinstance(df, nw.LazyFrame)
+        else does_not_raise()
+    )
+    with context:
+        result = df.select(nw.col("a").unique())
+        expected = {"a": [1, 2]}
+        assert_equal_data(result, expected)
+
+
+def test_unique_expr_agg(
+    constructor: Constructor, request: pytest.FixtureRequest
+) -> None:
+    if any(x in str(constructor) for x in ("duckdb", "pyspark")):
+        request.applymarker(pytest.mark.xfail)
+    df = nw.from_native(constructor(data))
+    result = df.select(nw.col("a").unique().sum())
+    expected = {"a": [3]}
     assert_equal_data(result, expected)
 
 
+def test_unique_illegal_combination(constructor: Constructor) -> None:
+    df = nw.from_native(constructor(data))
+    with pytest.raises(LengthChangingExprError):
+        df.select((nw.col("a").unique() + nw.col("b").unique()).sum())
```

Comment:
an expression like this is so bizarre that I doubt we'd be cutting off many real use-cases...

Reply:
Agree! 😂